actions/dynamic.go

```diff
@@ -20,3 +22,26 @@ func DriveCollectorByStorageControllerVendor(vendor string, trace bool) DriveCol
 
 	return nil
 }
+
+func driveCapabilityCollectorByLogicalName(logicalName string, trace bool, collectors []DriveCapabilityCollector) DriveCapabilityCollector {
+	// when collectors are is passed in, return the collector based on the logical name
+	for _, collector := range collectors {
+		nvme, isNvmeCollector := collector.(*utils.Nvme)
+		if isNvmeCollector && strings.Contains(logicalName, "nvme") {
+			return nvme
+		}
+
+		hdparm, isHdparmCollector := collector.(*utils.Hdparm)
+		if isHdparmCollector && !strings.Contains(logicalName, "nvme") {
+			return hdparm
+		}
+	}
+
+	// otherwise return the collector based on the logical name
```

Comment:
This is a safe assumption for our environment as discussed, but just noting it would need to be adjusted in the case of SAS or SCSI.  hdparm will at least return some basic info on SAS drives so it shouldn't fail outright.